fred2/volumetricsdlg.cpp

```diff
@@ -0,0 +1,391 @@
+#include "stdafx.h"
+#include "FRED.h"
+#include "volumetricsdlg.h"
+
+#include "nebula/volumetrics.h"
+
+#define ID_AND_SPIN(name) IDC_##name, IDC_SPIN_##name
+
+static constexpr char* Model_file_ext = "Model Files (*.pof)|*.pof||";
+static constexpr std::initializer_list<int> Interactible_fields = {
+	IDC_HULL, IDC_SET_HULL,
+	ID_AND_SPIN(POS_X),
+	ID_AND_SPIN(POS_Y),
+	ID_AND_SPIN(POS_Z),
+	ID_AND_SPIN(COLOR_R),
+	ID_AND_SPIN(COLOR_G),
+	ID_AND_SPIN(COLOR_B),
+	ID_AND_SPIN(OPACITY),
+	ID_AND_SPIN(OPACITY_DISTANCE),
+	ID_AND_SPIN(STEPS),
+	ID_AND_SPIN(RESOLUTION),
+	ID_AND_SPIN(OVERSAMPLING),
+	ID_AND_SPIN(HGCOEFF),
+	ID_AND_SPIN(SUN_FALLOFF),
+	ID_AND_SPIN(STEPS_SUN),
+	ID_AND_SPIN(EM_SPREAD),
+	ID_AND_SPIN(EM_INTENSITY),
+	ID_AND_SPIN(EM_FALLOFF),
+	IDC_NOISE_ENABLE
+};
+static constexpr std::initializer_list<int> Interactible_noise_fields = {
+	ID_AND_SPIN(NOISE_COLOR_R),
+	ID_AND_SPIN(NOISE_COLOR_G),
+	ID_AND_SPIN(NOISE_COLOR_B),
+	ID_AND_SPIN(NOISE_SCALE_B),
+	ID_AND_SPIN(NOISE_SCALE_S),
+	ID_AND_SPIN(NOISE_INTENSITY),
+	ID_AND_SPIN(NOISE_RESOLUTION),
+	//IDC_NOISE_BASE, IDC_NOISE_SUB //ToDo: Add once buttons are implemented
+};
+
+volumetrics_dlg::volumetrics_dlg(CWnd* pParent /*=nullptr*/) : CDialog(volumetrics_dlg::IDD, pParent),
+	m_enabled(false),
+	m_volumetrics_hull("hull_pof"),
+	m_position(ZERO_VECTOR),
+	m_color({255, 255, 255}), 
+	m_opacity(0.001f),
+	m_distance(5.0f),
+	m_steps(15),
+	m_resolution(6),
+	m_oversampling(2),
+	m_henyeyGreenstein(0.2f),
+	m_sunFalloffFactor(1.0f),
+	m_sunSteps(6),
+	m_emissiveSpread(0.7f),
+	m_emissiveIntensity(1.1f),
+	m_emissiveFalloff(1.5f),
+	m_noise(false),
+	m_noisecolor({0, 0, 0}),
+	m_noiseScaleBase(25.0f),
+	m_noiseScaleSub(14.0f),
+	m_noiseIntensity(1.0f),
+	m_noiseResolution(5)
+{}
+
+volumetrics_dlg::~volumetrics_dlg()
+{
+}
+
+BOOL volumetrics_dlg::OnInitDialog()
```

Comment:
I think the first line of this function should be a call to `CDialog::OnInitDialog();`